build/js/dropkick.min.js

```diff
@@ -0,0 +1 @@
tion-highlight")&&this.select(n);case o.esc:this.isOpen&&(e.preventDefault(),this.close())}},_searchOptions:function(e){var t,i=this,a=String.fromCharCode(e.keyCode||e.which),n=function(){i.data.searchTimeout&&clearTimeout(i.data.searchTimeout),i.data.searchTimeout=setTimeout(function(){i.data.searchString=""},1e3)};this.data.searchString===s&&(this.data.searchString=""),n(),this.data.searchString+=a,t=this.search(this.data.searchString,this.data.settings.search),t.length&&(h.hasClass(t[0],"dk-option-disabled")||this.selectOne(t[0]))},_scrollTo:function(e){var t,s,i,a=this.data.elem.lastChild;return-1===e||"number"!=typeof e&&!e||!this.isOpen&&!this.multiple?!1:("number"==typeof e&&(e=this.item(e)),t=h.position(e,a).top,s=t-a.scrollTop,i=s+e.offsetHeight,void(i>a.offsetHeight?(t+=e.offsetHeight,a.scrollTop=t-a.offsetHeight):0>s&&(a.scrollTop=t)))}},l.build=function(e,t){var s,i,a,n=[],o={elem:null,options:[],selected:[]},l=function(e){var s,i,a,n,r=[];switch(e.nodeName){case"OPTION":s=h.create("li",{"class":"dk-option ","data-value":e.value,innerHTML:e.text,role:"option","aria-selected":"false",id:t+"-"+(e.id||e.value.replace(" ","-"))}),h.addClass(s,e.className),e.disabled&&(h.addClass(s,"dk-option-disabled"),s.setAttribute("aria-disabled","true")),e.selected&&(h.addClass(s,"dk-option-selected"),s.setAttribute("aria-selected","true"),o.selected.push(s)),o.options.push(this.appendChild(s));break;case"OPTGROUP":for(i=h.create("li",{"class":"dk-optgroup"}),e.label&&i.appendChild(h.create("div",{"class":"dk-optgroup-label",innerHTML:e.label})),a=h.create("ul",{"class":"dk-optgroup-options"}),n=e.children.length;n--;r.unshift(e.children[n]));r.forEach(l,a),this.appendChild(i).appendChild(a)}};for(o.elem=h.create("div",{"class":"dk-select"+(e.multiple?"-multi":"")}),i=h.create("ul",{"class":"dk-select-options",id:t+"-listbox",role:"listbox"}),e.disabled&&h.addClass(o.elem,"dk-select-disabled"),o.elem.id=t+(e.id?"-"+e.id:""),h.addClass(o.elem,e.className),e.multiple?(o.elem.setAttribute("tabindex",e.getAttribute("tabindex")||"0"),i.setAttribute("aria-multiselectable","true")):(s=e.options[e.selectedIndex],o.elem.appendChild(h.create("div",{"class":"dk-selected "+s.className,tabindex:e.tabindex||0,innerHTML:s?s.text:"&nbsp;",id:t+"-combobox","aria-live":"assertive","aria-owns":i.id,role:"combobox"})),i.setAttribute("aria-expanded","false")),a=e.children.length;a--;n.unshift(e.children[a]));return n.forEach(l,o.elem.appendChild(i)),o},l.onDocClick=function(e){var t,s;if(1!==e.target.nodeType)return!1;null!==(t=e.target.getAttribute("data-dkcacheid"))&&l.cache[t].focus();for(s in l.cache)h.closest(e.target,l.cache[s].data.elem)||s===t||l.cache[s].disabled||l.cache[s].close()},e.Dropkick=l,e.jQuery!==s&&(e.jQuery.fn.dropkick=function(){var t=Array.prototype.slice.call(arguments);return e.jQuery(this).each(function(){t[0]&&"object"!=typeof t[0]?"string"==typeof t[0]&&l.prototype[t[0]].apply(new l(this),t.slice(1)):new l(this,t[0]||{})})})}(window,document);
```

Comment:
Line is too long.<br>'e' is already defined.<br>Missing semicolon.<br>Expected an assignment or function call and instead saw an expression.<br>Wrap an immediate function invocation in parens to assist the reader in understanding that the expression is the result of a function, and not the function itself.<br>Use '===' to compare with 'null'.<br>Expected '{' and instead saw 'throw'.<br>Unexpected use of '>>>'.<br>Mixed double and single quotes.<br>Expected '{' and instead saw 'return'.<br>A constructor name should start with an uppercase letter.<br>Expected '{' and instead saw 'if'.<br>Expected '{' and instead saw 'for'.<br>Expected '{' and instead saw 'e'.<br>The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.<br>Expected '{' and instead saw 's'.<br>Too many errors. (100% scanned).